bridge_adaptivity/api/backends/openedx.py

```diff
@@ -117,71 +119,104 @@ def get_provider_courses(self, username=None, org=None, mobile=None):
             org=org,
             mobile=mobile
         )
-        return resource.get('results')
+        return filter_function(resource.get('results'))
 
 
-def get_available_blocks(course_id):
+def check_source_id(source_id, not_allow_empty_source_id=True):
+    """
+    Check that passed source_id parameter is valid.
+
+    If there's only one active source provider - source_id parameter is not required, it will get first active.
+    :param source_id: LtiConsumer object id
+    :return: queryset of content_sources
+    :raise HttpClientError: if provided parameters are not valid
+    """
+    content_sources = get_content_providers(source_id=source_id)
+    if not content_sources:
+        # if no active content sources
+        raise HttpClientError(_("No active Content Provider"))
+
+    if not_allow_empty_source_id:
+        require_source_id = content_sources.count() > 1
+        if require_source_id and not source_id:
+            # if source_id is not provided and more than one active content_provider
+            raise HttpClientError(_("Parameter source_id is mandatory if there's more than one active content source."))
+
+    return content_sources
+
+
+def get_available_blocks(source_id=None, course_id=''):
     """
     Content Source API requester.
 
-    Fetches all source blocks from the course with given ID.
+    Fetches all source blocks from the course with the given ID.
     Blocks data is filtered by `apply_data_filter`.
-    :param course_id:
+    :param course_id: Course id
+    :param source_id: LtiConsumer id
     :return: (list) blocks data
     """
-    content_source = get_content_provider()
-    if not content_source:
-        raise HttpClientError(_("No active Content Provider"))
-
-    # Get API client instance:
-    api = OpenEdxApiClient(content_source=content_source)
+    content_sources = check_source_id(source_id)
```

Comment:
We did API call for available blocks every time we change the course, why do we need to get through all content sources if we are interested in the certain source and course?